src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/DynamicFeatureKeyValueAdapter.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Data.AppConfiguration;
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement
+{
+    internal class DynamicFeatureKeyValueAdapter : IKeyValueAdapter
+    {
+        public Task<IEnumerable<KeyValuePair<string, string>>> ProcessKeyValue(ConfigurationSetting setting, CancellationToken cancellationToken)
+        {
+            DynamicFeature dynamicFeature;
+
+            try
+            {
+                 dynamicFeature = JsonSerializer.Deserialize<DynamicFeature>(setting.Value);
+            }
+            catch (JsonException e)
+            {
+                throw new FormatException(setting.Key, e);
+            }
+
+            var keyValues = new List<KeyValuePair<string, string>>();
+
+            keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.FeatureManagementSectionName}:{FeatureManagementConstants.DynamicFeaturesSectionName}:{dynamicFeature.Id}:Assigner", dynamicFeature.ClientAssigner));
+
+            for (int i = 0; i < dynamicFeature.Variants.Count; i++)
+            {
+                FeatureVariant variant = dynamicFeature.Variants[i];
+
+                keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.FeatureManagementSectionName}:{FeatureManagementConstants.DynamicFeaturesSectionName}:{dynamicFeature.Id}:Variants:{i}:Name", variant.Name));
```

Comment:
Instead of ":" I'd recommend we use `ConfigurationPath.KeyDelimiter`.

Reply:
Its making the strings longer and harder to read if we use `ConfigurationPath.KeyDelimiter`. For example:
```
string variantsSectionPrefix = DynamicFeatureSectionPrefix +
                               ConfigurationPath.KeyDelimiter +
                               dynamicFeature.Id +
                               ConfigurationPath.KeyDelimiter + 
                               FeatureManagementConstants.Variants +
                               ConfigurationPath.KeyDelimiter + 
                               i;
```

Vs using the `:` character:

```
string variantsSectionPrefix = $"{DynamicFeatureSectionPrefix}:{dynamicFeature.Id}:{FeatureManagementConstants.Variants}:{i}";

```

Reply:
Is the concern that `ConfigurationPath.KeyDelimiter` is too big and little but unwieldy? In that case, should be able to do one of the 2.

```
using Delimiter = ConfigurationPath.KeyDelimiter;
```

or 
```
private readonly string Delimiter = ConfigurationPath.KeyDelimiter;
```

Reply:
I was looking to avoid repeated use of literals.

Reply:
Even if we use a shorter name to represent this literal, I think using the `:` character makes the string very easy to read and understand.
For example, which looks more natural to you?
This -
```
            keyValues.Add(
                new KeyValuePair<string, string>(
                    DynamicFeatureSectionPrefix +
                    Delimiter +
                    dynamicFeature.Id +
                    Delimiter +
                    FeatureManagementConstants.DynamicFeatureAssigner,
                    dynamicFeature.ClientAssigner));

```

Or this -
```
            keyValues.Add(
                new KeyValuePair<string, string>(
                    $"{DynamicFeatureSectionPrefix}:{dynamicFeature.Id}:{FeatureManagementConstants.Assigner}",
                    dynamicFeature.ClientAssigner));
```

Are there any side effects of repeatedly using the `:` string literal? If yes, another option is to use `string.join()` which IMO makes the code easier to read, but that has an overhead of creating a string array containing the values to be joined.

Reply:
There's no side effects. It's the same reason to avoid typing out the literal value of `DynamicFeatureSectionPrefix` each time.  More of a common practice to avoid repeated use of literals. If the literal needs to change for some reason it can be changed at the source rather than at every occurrence. 

If you see value in using the colon directly and want to keep it that way then you can resolve this.